+adi/+Generic/Rx.m

```diff
@@ -0,0 +1,237 @@
+    %   data from any generic ADI precision ADC.
+    %
+    %   rx = adi.Generic.Rx('adxxxx', 'ip:analog.local');
+    %
+
+    properties
+        % DeviceAttributeNames Device Attribute Names
+        %   Array of names device IIO attributes as defined in the
+        %   driver/firmware, and queried from the hardware
+        %   at initialization time
+        DeviceAttributeNames
+
+        % ChannelAttributeNames Device Attribute Names
+        %   Array of names channel IIO attributes as defined in the
+        %   driver/firmware, and queried from the hardware
+        %   at initialization time
+        ChannelAttributeNames
+    end
+
+    properties (Nontunable)
+        % SamplesPerFrame Samples Per Frame
+        %   Number of samples per frame, specified as an even positive
+        %   integer.
+        SamplesPerFrame = 1024
+    end
+
+    properties (Nontunable, Hidden)
+        channel_names = {'voltage0'}
+        Timeout = Inf
+        kernelBuffersCount = 1
+        dataTypeStr = 'int32'
+        phyDevName = 'adxxxx'
+        devName = 'adxxxx'
+    end
+
+    properties (Nontunable, Hidden, Constant)
+        Type = 'Rx'
+        ComplexData = false
+    end
+
+    properties (Hidden, Nontunable, Access = protected)
+        isOutput = false
+    end
+
+    properties(Hidden, Access = protected)
+        cachedAttrWrites = {}
+    end
+
+    methods
+
+        %% Constructor
+        function obj = Rx(varargin)
+            obj = obj@matlabshared.libiio.base(varargin{4:end});
+            obj.enableExplicitPolling = false;
+            obj.EnabledChannels = 1;
+            obj.BufferTypeConversionEnable = true;
+
+            % Set device name and uri from input arguments
+            obj.devName = varargin{1};
+            obj.phyDevName = varargin{1};
+            obj.uri = varargin{2};
+
+            % Determine datatype using trial and error
+            possibleDataTypeStr = {'int8', 'int16', 'int32', 'int64'};
+            for k = 1:length(possibleDataTypeStr)
+                try
+                    obj.dataTypeStr = possibleDataTypeStr{k};
+                    obj.setup();
+                    break;
+                catch
+                end
+            end
+            obj.dataTypeStr
+            assert(obj.ConnectedToDevice == 1, "Connection could not be established with the specified device")
+
+            % Check is the platform is running ADI Kuiper Linux.
+            % Presence of 'local,kernel' context attribute would
+            % indicate usage of Linux platform
+            isLinuxPLatform = false;
+            try
+                obj.iio_context_get_attr_value(obj.iioCtx, 'local,kernel');
+                isLinuxPLatform = true;
+            catch
+            end            
+
+            release(obj);
+
+            % If platform is running ADI Kuiper Linux, kernelBuffersCount
+            % can be increased
+            if isLinuxPLatform
+                obj.kernelBuffersCount = 4;
```

Comment:
We could do that I guess, but it would get executed multiple times without any benefit..

Just to confirm, are you suggesting it so that we check if the value for `kernelBuffersCount` is valid by doing the setup? If so, we can add a check here itself ig?

Reply:
No, I was just suggesting to add in same if condition but looks like you are saying it will be executed more than once. Lets' leave as-is then